File: backend/spring/src/main/java/io/ssafy/openticon/controller/response/PackDownloadResponseDto.java
package io.ssafy.openticon.controller.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Getter
@RequiredArgsConstructor
@AllArgsConstructor
public class PackDownloadResponseDto {

    private Long id;
    private String title;
    private boolean isPublic;
    private String uuid;
    private String thumbnailImg;
    private String listImg;
    private List<String> emoticons;


}

File: android/app/src/main/java/io/ssafy/openticon/data/model/PackDownloadResponseDto.kt
package io.ssafy.openticon.data.model

import io.ssafy.openticon.domain.model.EmoticonPack

data class PackDownloadResponseDto(
    val thumbnailImg: String,
    val listImg: String,
    val emoticons: List<String>,
    val id: Int,
    val title: String,
    val isPublic: Boolean,
    val uuid: String
) {
    fun toEmoticonPack(): EmoticonPack {
        return EmoticonPack(
            thumbnail = thumbnailImg,
            listImg = listImg,
            emoticonUrls = emoticons,
            id = id,
            title = title,
            isPublic = isPublic,
            uuid = uuid
        )
    }
}